Update docs for interleaved and implement integration tests. (#185)
GoogleCloudPlatform/google-cloud-spanner-hibernate: README.adoc

Hibernate performs schema updates on each table and entity type on startup, which can take more than several minutes if there are many tables. To avoid schema updates keeping Hibernate from starting for several minutes, you can update schemas separately and use the `none` or `validate` settings.

==== Interleaved Tables

> Cloud Spanner offers the concept of https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables[Interleaved Tables] which allows you to cluster the rows of an interleaved table with rows of a parent table.

Reviewer: ```suggestion
Cloud Spanner offers the concept of https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables[Interleaved Tables] which allows you to co-locate the rows of an interleaved table with rows of a parent table for efficient retrieval.
```

> This feature enforces the one-to-many relationship and provide efficient queries and operations on entities of a single domain parent entity.

Reviewer: ```suggestion
This feature enforces the one-to-many relationship and provides efficient queries and operations on entities of a single domain parent entity.
```

> If you would like to generate interleaved tables in Spanner, you must annotate your entity with the `@Interleaved` annotation.

Reviewer: ```suggestion
If you would like to generate interleaved tables in Cloud Spanner, you must annotate your entity with the `@Interleaved` annotation.
```
